extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeployer.java

```diff
@@ -250,6 +273,32 @@ private void printExposeInformation(KubernetesClient client, KubernetesList list
         }
     }
 
+    /**
+     * Obtain everything the APIResourceList from the server and extract all the info we need in order to know how to create /
+     * delete the specified generic resource.
+     * 
+     * @param client the client instance to use to query the server.
+     * @param resource the generic resource.
+     * @return an optional {@link ResourceDefinitionContext} with the resource info or empty if resource could not be matched.
+     */
+    private static Optional<ResourceDefinitionContext> getGenericResourceContext(KubernetesClient client,
+            GenericKubernetesResource resource) {
+        APIResourceList apiResourceList = client.getApiResources(resource.getApiVersion());
+        if (apiResourceList == null || apiResourceList.getResources() == null || apiResourceList.getResources().isEmpty()) {
+            return Optional.empty();
+        }
+        return client.getApiResources(resource.getApiVersion()).getResources().stream()
+                .filter(r -> r.getKind().equals(resource.getKind()))
+                .map(r -> new ResourceDefinitionContext.Builder()
+                        .withGroup(ApiVersionUtil.trimGroup(resource.getApiVersion()))
+                        .withVersion(ApiVersionUtil.trimVersion(resource.getApiVersion()))
+                        .withKind(r.getKind())
+                        .withNamespaced(r.getNamespaced())
```

Comment:
So the whole point is to figure out whether the resource is namespaced and what its plural is? Would it make sense to first try the operation with some assumptions (i.e. that the resource is namespaced and what its plural is (using fabric8's `Pluralize`) and only query the cluster for the information if that failed? Or we could even try twice: once namespaced, once cluster and finally query the cluster if it still doesn't work? 
Not sure what's the cluster impact performance-wise of enumerating all resources… 
LGTM otherwise.

Reply:
This logic is already present in the client:

https://github.com/fabric8io/kubernetes-client/blob/7ef49cd34e2de0d68422e17ce24fe15e48b63f17/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Handlers.java#L83

It can be reused by using `client#resources(Class<T> resourceType, Class<L> listClass)` or `client#genericKubernetesResources(String apiVersion, String kind)` 

I'm not sure if you are aware of this.